pennylane_lightning/core/src/bindings/BindingsLTensor.hpp

```diff
@@ -0,0 +1,321 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file BindingsLTensor.hpp
+ * Defines device-agnostic operations to export to Python and other utility
+ * functions interfacing with Pybind11.
+ */
+
+#pragma once
+#include <set>
+#include <string>
+#include <string_view>
+#include <tuple>
+#include <vector>
+
+#include <pybind11/complex.h>
+#include <pybind11/functional.h>
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+
+#include "CPUMemoryModel.hpp" // CPUMemoryModel, getMemoryModel, bestCPUMemoryModel, getAlignment
+#include "Macros.hpp" // CPUArch
+#include "Util.hpp"   // for_each_enum
+
+#include "BindingsLTensorBase.hpp"
+#include "LTensorTNCudaBindings.hpp"
+#include "MeasurementsTNCuda.hpp"
+#include "ObservablesTNCuda.hpp"
+
+namespace py = pybind11;
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::LightningTensor::TNCuda;
+using namespace Pennylane::LightningTensor::TNCuda::Observables;
+using namespace Pennylane::LightningTensor::TNCuda::Measures;
+} // namespace
+/// @endcond
+
+namespace Pennylane {
+/**
+ * @brief Return basic information of the compiled binary.
+ */
+auto getCompileInfo() -> py::dict {
+    using namespace Pennylane::Util;
+    using namespace py::literals;
+
+    const std::string_view cpu_arch_str = [] {
+        switch (cpu_arch) {
+        case CPUArch::X86_64:
+            return "x86_64";
+        case CPUArch::PPC64:
+            return "PPC64";
+        case CPUArch::ARM:
+            return "ARM";
+        default:
+            return "Unknown";
+        }
+    }();
+
+    const std::string_view compiler_name_str = [] {
+        switch (compiler) {
+        case Compiler::GCC:
+            return "GCC";
+        case Compiler::Clang:
+            return "Clang";
+        case Compiler::MSVC:
+            return "MSVC";
+        case Compiler::NVCC:
+            return "NVCC";
+        case Compiler::NVHPC:
+            return "NVHPC";
+        default:
+            return "Unknown";
+        }
+    }();
+
+    const auto compiler_version_str = getCompilerVersion<compiler>();
+
+    return py::dict("cpu.arch"_a = cpu_arch_str,
+                    "compiler.name"_a = compiler_name_str,
+                    "compiler.version"_a = compiler_version_str,
+                    "AVX2"_a = use_avx2, "AVX512F"_a = use_avx512f);
+}
+
+/**
+ * @brief Register bindings for general info.
+ *
+ * @param m Pybind11 module.
+ */
+void registerInfo(py::module_ &m) {
+    /* Add compile info */
+    m.def("compile_info", &getCompileInfo, "Compiled binary information.");
+}
+
+/**
+ * @brief Register observable classes.
+ *
+ * @tparam StateTensorT
+ * @param m Pybind module
+ */
+template <class StateTensorT>
+void registerBackendAgnosticObservables(py::module_ &m) {
```

Comment:
Can we use the one that is implemented for state-vector devices?

Reply:
It's technically feasible with more conditional compilation directives or keep this part in a seperate translate unit. I would like to keep the bindings as separate as possible for now and we can come back to it later once we get all features in.